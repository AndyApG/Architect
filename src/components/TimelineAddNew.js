import React from 'react';
import PropTypes from 'prop-types';
import { Icon, colorDictionary } from 'network-canvas-ui';
import { Zoom } from '../behaviours';

const zoomColors = [colorDictionary.background, colorDictionary['light-background']];
const constrain = [60, 0, 0, 0];

const AddNewButton = Zoom(
  ({ onInsertStage }) => (
    <div
      className="timeline-add-new__button"
      role="button"
      tabIndex="0"
      onClick={onInsertStage}
      style={{ display: 'inline-block' }}
    >
      <div className="timeline-add-new__button-label">Add new stage here</div>
      <Icon name="add-a-screen" className="timeline-add-new__button-icon" />
    </div>
  ),
);

const TimelineAddNew = ({ onInsertStage }) => (
  <div className="timeline-add-new">
    <AddNewButton
      zoomColors={zoomColors}
      onInsertStage={onInsertStage}
      constrain={constrain}
    />
  </div>
);

TimelineAddNew.propTypes = {
  onInsertStage: PropTypes.func.isRequired,
};

export default TimelineAddNew;
