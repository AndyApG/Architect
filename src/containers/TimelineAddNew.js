/*eslint-disable*/

import React from 'react';
import PropTypes from 'prop-types';
import { Icon, colorDictionary as colors } from 'network-canvas-ui';
import { ZoomElement } from '../components';

const zoomColors = [colors.background, colors['light-background']];

const TimelineAddNew = ({ onInsertStage }) => (
  <div class="timeline-add-new">
    <ZoomElement colors={zoomColors}>
      <div
        className="timeline-add-new__button"
        role="button"
        tabIndex="0"
        onClick={onInsertStage}
        style={{ display: 'inline-block' }}
      >
        <div className="timeline-add-new__button-label">Add new stage here</div>
        <Icon name="add-a-screen" className="timeline-add-new__button-icon" />
      </div>
    </ZoomElement>
  </div>
);

TimelineAddNew.propTypes = {
  onInsertStage: PropTypes.func.isRequired,
};

export default TimelineAddNew;
