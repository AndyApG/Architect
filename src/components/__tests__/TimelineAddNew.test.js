/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import TimelineAddNew from '../TimelineAddNew';

const mockProps = {
  onInsertStage: () => {},
};

describe('<TimelineAddNew />', () => {
  it('can render', () => {
    const component = shallow(<TimelineAddNew {...mockProps} />);

    expect(component).toMatchSnapshot();
  });
});
