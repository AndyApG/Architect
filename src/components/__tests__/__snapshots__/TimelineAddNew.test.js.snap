// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineAddNew /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineAddNew
    onInsertStage={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <getContext(Draft(Component))
        onInsertStage={[Function]}
        zoomColors={
                Array [
                        "#3a3a75",
                        "#3a3a75",
                      ]
        }
/>,
      "className": "timeline-add-new",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "onInsertStage": [Function],
        "zoomColors": Array [
          "#3a3a75",
          "#3a3a75",
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <getContext(Draft(Component))
          onInsertStage={[Function]}
          zoomColors={
                    Array [
                              "#3a3a75",
                              "#3a3a75",
                            ]
          }
/>,
        "className": "timeline-add-new",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "onInsertStage": [Function],
          "zoomColors": Array [
            "#3a3a75",
            "#3a3a75",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
